torch/sobolengine.py

```diff
@@ -0,0 +1,47 @@
+import torch
+
+class SobolEngine(object):
+
+    MAXBIT = 30
+
+    def __init__(self, dimension, scramble=False, seed=None):
+        if dimen > 1111 or dimen < 1:
+            raise ValueError("Supported range of dimensionality for SobolEngine is [1, 1111]")
+
+        self.seed = seed
+        self.scramble = scramble
+        self.dimension = dimension
+
+        self.sobolstate = torch.zeros(dimension, self.MAXBIT, dtype=torch.int)
+        torch._sobol_engine_initialize_state(self.sobolstate, self.dimension)
+
+        if scramble:
+            self.shift = torch.mv(torch.randint(2, (self.dimension, self.MAXBIT)),
```

Comment:
There are two ways of seeding: using a `torch.Generator` object and then seeding to, so that the seeding is "local". Or you could modify the global seed as well.

Clearly, option 1 is the better approach, but I am not sure about it. There is no documentation of how to obtain `torch.Generator` objects to pass to random tensor factory functions. Once I work out the details, I will add it.

Reply:
This has been taken care of. Scrambling should work now.